server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -649,16 +648,6 @@ TabletGoalState getGoalState(TabletLocationState tls, MergeInfo mergeInfo) {
               break;
             case STARTED:
             case SPLITTING:
-              return TabletGoalState.HOSTED;
-            case WAITING_FOR_CHOPPED:
-              if (tls.getState(tserverSet.getCurrentServers()).equals(TabletState.HOSTED)) {
-                if (tls.chopped) {
-                  return TabletGoalState.UNASSIGNED;
-                }
-              } else if (tls.chopped && tls.walogs.isEmpty()) {
-                return TabletGoalState.UNASSIGNED;
-              }
-
```

Comment:
One of the things I wasn't sure about was if we were just going to leave all the chop compaction related stuff around as deprecated even if not used anymore or just delete it all. I figured removing may cause issues with upgrades or other problems but I wasn't sure since chop compaction code should mostly be internal in the servers and I wouldn't expect to touch public API a lot (but I haven't checked), although there is a Thrift api for running a chop compaction. In the elasticity branch I figure we will delete it all but in main I guess we can keep it and mark as deprecated.

I was going to tackle the deprecation/removal in a follow on PR (upgrading metadata would also be a follow on PR) so I can revert back the things I removed such as this switch statement for now. We also need to handle the upgrade case in a follow on PR to upgrade existing metadata to the new format.

Reply:
Definitely delete what isn't needed, if it's internal code. The only thing you need to be careful about keeping around is stuff related to serializing (if it's persisted anywhere needed across upgrades), or if it's public API. Then, we need to be more careful.

Reply:
> Unfortunately, we (inconveniently) chose to use the ordinal to serialize the merge info for fate operations, so we have to be very careful not to cause a problem by changing the ordinals and interpreting them incorrectly across an upgrade. If we hadn't done that (which, in general, is a bad idea), then we could have just removed it entirely, since it's only used internally, and we expect all the jars we use to be upgraded at the same time.

Accumulo used to not upgrade if there are outstanding FATE ops. Reasoning about FATEs operations running across an upgrade is not easy to do, its also really difficult to test.  So we should be able to remove this enum.  If Accumulo is no longer preventing upgrade when there are outstanding FATE ops, then I would consider that a bug.

Reply:
Found the code that does the check for outstanding fate ops in the upgrade code.

https://github.com/apache/accumulo/blob/770c05b46a4e3c135c85d0772ebb2b02d9db5414/server/manager/src/main/java/org/apache/accumulo/manager/upgrade/UpgradeCoordinator.java#L157